modules/oracledb/oracledb.go

```diff
@@ -0,0 +1,629 @@
+package oracledb
+
+import (
+	"database/sql"
+	"strconv"
+	"strings"
+
+	"github.com/netdata/go.d.plugin/modules"
+
+	"gopkg.in/goracle.v2"
+)
+
+const defaultDSN = "SYSTEM/Oracle12345@ORCL" // there is no such thing as default connstr/dsn, but...
+
+func init() {
+	creator := modules.Creator{
+		// TODO: open a discussion about API changes with @ilyam for that creator thing and more.
+		Create: func() modules.Module { return New(defaultDSN) },
+	}
+
+	modules.Register("oracledb", creator)
+}
+
+// OracleDB oracledb module.
+type OracleDB struct {
+	modules.Base
+	DSN string `json:"dsn,omitempty" yaml:"dsn,omitempty"`
```

Comment:
Why do we need `json`? We use `yaml` for config files.

Reply:
✅